src/components/UnprovisionedSetup.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { FC, useState } from 'react';
+import TenantApiSetupForm from 'components/TenantAPISetupForm';
+import { DEFAULT_API_HOST } from 'components/constants';
+import { InstanceSelection } from 'components/InstanceSelection';
+import { RegistrationInfo } from 'types';
+import { Container } from '@grafana/ui';
+import {
+  getOrCreateMetricAndLokiDatasources,
+  initializeSMDatasource,
+  initializeTenant,
+  saveTenantInstanceIds,
+  updatePluginJsonData,
+} from 'fetchers';
+
+interface ApiSetupValues {
+  adminApiToken: string;
+  apiHost?: string;
+}
+
+interface Props {
+  pluginId: string;
+  pluginName: string;
+}
+
+/* 
+This component exists to facilitate bootstrapping the app from an unprovisioned state (i.e. a Grafana instance running on prem)
+We need to do the following things in the setup process:
+1. Post an admin token to the proxied init endpoint
+2. We receive back a list of instances hosted in Grafana Cloud and an accessToken from the init endpoint
+3. Present the instances for selection by the user
+4. Once the desired instances are selected, we need to create the SM datasource with the accessToken and SM API url in the jsonData so it can be used in subsequent requests to the SM API
+5. Ensure there are datasources for the selected metrics and logs instances (if not, create them)
+6. Store metadata about the metrics and logs instances in the SM datasource
+7. Import all the dashboards (the name of the newly created logs/metrics datasources is required for this)
+8. Send the selected metric and logs instance metadata to the SM API (with the register/save endpoint)
+9. Reload the page and let the routing take control
+*/
+
+export const UnprovisionedSetup: FC<Props> = ({ pluginId, pluginName }) => {
+  const [apiSetup, setApiSetup] = useState<ApiSetupValues | undefined>();
+  const [apiSetupError, setApiSetupError] = useState<string | undefined>();
+  const [tenantInfo, setTenantInfo] = useState<RegistrationInfo | undefined>();
+  const [instanceSelectionError, setInstanceSelectionError] = useState<string | undefined>();
+
+  const onSetupSubmit = async (setupValues: ApiSetupValues) => {
+    // put api host in plugin jsonData
+    try {
+      await updatePluginJsonData(pluginId, { apiHost: apiSetup?.apiHost ?? DEFAULT_API_HOST });
+      const tenantInfo = await initializeTenant(pluginId, setupValues.adminApiToken);
+      setTenantInfo(tenantInfo);
+      setApiSetup(setupValues);
+    } catch (e) {
+      setApiSetupError(e.message);
+    }
+  };
+
+  const onInstanceSelectionSubmit = async (
+    metricsInstanceId: number | undefined,
+    logsInstanceId: number | undefined
```

Comment:
**Suggestion:** probably we could use optional args here?

By using `metricsInstanceId?: number` instead of `metricsInstanceId: number | undefined`.